src/java/com/wolfssl/provider/jsse/WolfSSLSocket.java

```diff
@@ -1297,6 +1297,20 @@ public synchronized SSLSession getSession() {
             /* Log error, but continue. Session returned will be empty */
             WolfSSLDebug.log(getClass(), WolfSSLDebug.INFO,
                 "Handshake attempt failed in SSLSocket.getSession()");
+
+            /* close SSLSocket */
+            if (this.socket != null && !this.socket.isClosed()) {
```

Comment:
Do we need this socket null/isClosed check?  Since we call `close()` anyways, doesn't that check for this internal to that method?